src/notebooks/controllers/kernelSource/notebookKernelSourceSelector.ts

```diff
@@ -433,24 +473,64 @@ export class NotebookKernelSourceSelector implements INotebookKernelSourceSelect
         state: MultiStepResult,
         token: CancellationToken
     ) {
-        // Kick off a refresh of Python environments when displaying the quick pick for local kernels or Python envs.
-        this.interpreterService.refreshInterpreters().ignoreErrors();
         state.source = source;
         const onDidChange = new EventEmitter<void>();
+        let recommended: KernelConnectionMetadata | undefined;
+        const onDidChangeRecommended = new EventEmitter<void>();
         const provider = {
             onDidChange: onDidChange.event,
             onDidChangeStatus: source.onDidChangeStatus,
+            onDidChangeRecommended: onDidChangeRecommended.event,
             get kernels() {
                 return source.kernels;
             },
             get status(): 'discovering' | 'idle' {
                 return source.status;
             },
+            get recommended() {
+                return recommended;
+            },
             refresh: () => source.refresh()
         };
         const disposable = source.onDidChangeKernels(() => onDidChange.fire());
         state.disposables.push(disposable);
         state.disposables.push(onDidChange);
+        state.disposables.push(onDidChangeRecommended);
+
+        if (
```

Comment:
Maybe this is a good time to split `selectKernelFromKernelFinder` to `selectKernelFromRemoteKernelFinder` and `selectKernelFromLocalKernelFinder`. It seems the local one is getting more complicated.

Reply:
Done.